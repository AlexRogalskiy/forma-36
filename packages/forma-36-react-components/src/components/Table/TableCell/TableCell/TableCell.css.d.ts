// This file is automatically generated by typings-for-css-modules.
// Please do not change this file!
export interface ITableCellCss {
  'TableCell': string;
  'TableCell--head': string;
  'TableCell--head__sorting': string;
}

export const locals: ITableCellCss;
