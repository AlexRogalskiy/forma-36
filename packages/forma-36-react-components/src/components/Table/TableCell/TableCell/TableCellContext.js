import React from 'react';

export const contextOptions = {
  body: { name: 'body', element: 'td' },
  head: { name: 'head', element: 'th' },
};

export const TableCellContext = React.createContext(contextOptions.body);
