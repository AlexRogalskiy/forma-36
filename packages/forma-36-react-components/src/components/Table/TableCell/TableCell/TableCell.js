import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import { TableCellContext } from './TableCellContext';

import styles from './TableCell.css';

export const sortingDirections = {
  asc: 'asc',
  desc: 'desc',
};

export default class TableCell extends React.Component {
  static propTypes = {
    extraClassNames: PropTypes.string,
    children: PropTypes.node,
    sorting: PropTypes.oneOf([false, ...Object.keys(sortingDirections)]),
    align: PropTypes.oneOf(['center', 'left', 'right']),
  };

  static defaultProps = {
    children: null,
    extraClassNames: '',
    sorting: false,
    align: 'left',
  };

  render() {
    const {
      extraClassNames,
      children,
      sorting,
      align,
      ...otherProps
    } = this.props;

    return (
      <TableCellContext.Consumer>
        {({ name: context, element: Element }) => (
          <Element
            className={cn(styles.TableCell, extraClassNames, {
              [styles['TableCell--head']]: context === 'head',
              [styles['TableCell--head__sorting']]: sorting,
            })}
            align={align}
            {...otherProps}
          >
            {children}
          </Element>
        )}
      </TableCellContext.Consumer>
    );
  }
}
