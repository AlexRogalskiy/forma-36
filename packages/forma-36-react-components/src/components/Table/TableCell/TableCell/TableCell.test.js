import React from 'react';
import { shallow } from 'enzyme';
import TableCell from './TableCell';

it('renders the component', () => {
  const output = shallow(
    <TableCell active direction="asc">
      Foo
    </TableCell>,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component inactive', () => {
  const output = shallow(<TableCell sorting="asc">Foo</TableCell>);

  expect(output).toMatchSnapshot();
});
