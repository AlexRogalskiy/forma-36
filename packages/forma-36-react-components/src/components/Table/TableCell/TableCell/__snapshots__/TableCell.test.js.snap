// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `<[object Object] />`;

exports[`renders the component inactive 1`] = `<[object Object] />`;
